src/analysis/object.rs

```diff
@@ -37,6 +52,42 @@ impl Info {
     pub fn has_action_signals(&self) -> bool {
         self.signals.iter().any(|s| s.action_emit_name.is_some())
     }
+
+    /// Returns the location of the function within this object
+    pub fn function_location(&self, fn_info: &functions::Info) -> LocationInObject {
+        if self.final_type
+            || matches!(
+                fn_info.kind,
+                FunctionKind::Constructor | FunctionKind::Function
+            )
+        {
+            LocationInObject::Impl
+        } else if fn_info.status == GStatus::Generate {
+            LocationInObject::Ext
+        } else {
+            LocationInObject::ExtManual
+        }
+    }
+
+    /// Generate doc name based on function location within this object
+    /// See also [`Self::function_location()`].
+    /// Returns `(item/crate path including type name, just the type name)`
+    pub fn generate_doc_link_info(&self, fn_info: &functions::Info) -> (String, String) {
+        match self.function_location(fn_info) {
+            LocationInObject::Impl => (self.name.clone(), self.name.clone()),
+            loc @ LocationInObject::Ext | loc @ LocationInObject::ExtManual => {
+                let type_name = if loc == LocationInObject::ExtManual {
+                    format!("{}Manual", self.trait_name)
+                } else {
+                    self.trait_name.clone()
+                };
+                (format!("prelude::{}", type_name), type_name)
+            }
```

Comment:
```suggestion
            LocationInObject::ExtManual => {
                (format!("prelude::{}Manual", self.trait_name), type_name)
            }
            LocationInObject::Ext => {
                (format!("prelude::{}", self.trait_name), trait_name)
            }
```

Reply:
@GuillaumeGomez I thought about the same, but ended up not doing that because `ExtManual` needs to format the type twice for them to be identical:

```suggestion
            LocationInObject::ExtManual => {
                (format!("prelude::{}Manual", self.trait_name), /* HERE: */ format!("{}Manual", type_name))
            }
            LocationInObject::Ext => {
                (format!("prelude::{}", self.trait_name), trait_name)
            }
```

Your suggestion confirms that that's hard to miss, and that it might have been a good choice to leave this as-is :grimacing:

Reply:
Except that the current code is very hard to read and is cloning unnecessarily. In addition: we match against `LocationInObject::ExtManual`, so it seems pretty obvious. At least much more than having binding and then inner condition which is completely unnecessary.

Reply:
Yes 2x `loc @` is horrible, and putting the `clone` behind a `Cow` is nasty too. All the `format!`s do allocations too though. FIne to change this then.

Reply:
Thanks! :)

Reply:
How about doing `Manual` formatting once though:
```
            LocationInObject::ExtManual => {
                let trait_name = format!("{}Manual", self.trait_name);
                (format!("prelude::{}", trait_name), trait_name)
            }
            LocationInObject::Ext => {
                (format!("prelude::{}", self.trait_name), self.trait_name.clone())
            }
```

Your suggestion forget to `clone` in `::Ext` btw.

Reply:
You could return a `(String, &str)`, instead, no?

Reply:
The latter would have to be a `Cow<'static, str>` so that `ExtManual` can pass the _owned_ result of `format!`.

Reply:
@GuillaumeGomez Would this work for you?:

```rust
    pub fn generate_doc_link_info(
        &self,
        fn_info: &functions::Info,
    ) -> (Cow<'_, str>, Cow<'_, str>) {
        match self.function_location(fn_info) {
            LocationInObject::Impl => (self.name.as_str().into(), self.name.as_str().into()),
            LocationInObject::ExtManual => {
                let trait_name = format!("{}Manual", self.trait_name);
                (format!("prelude::{}", trait_name).into(), trait_name.into())
            }
            LocationInObject::Ext => (
                format!("prelude::{}", self.trait_name).into(),
                self.trait_name.as_str().into(),
            ),
            LocationInObject::Builder => {
                panic!("C documentation is not expected to link to builders (a Rust concept)!")
            }
        }
    }
}
```

Reply:
Yep it's fine. :)

Reply:
@bilelmoussaoui You know what to copy-paste :wink: